src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/RawBinaryReader.cpp

```diff
@@ -0,0 +1,245 @@
+{
+  if(fileSize - skipHeaderBytes < allocatedBytes)
+  {
+    return -1;
+  }
+  if(fileSize - skipHeaderBytes > allocatedBytes)
+  {
+    return 1;
+  }
+  // File Size and Allocated Size are equal so we are good to go
+  return 0;
+}
+
+// -----------------------------------------------------------------------------
+template <typename T>
+int32_t readBinaryFile(IDataArray* dataArrayPtr, const std::string& filename, uint64_t skipHeaderBytes, ChoicesParameter::ValueType endian)
+{
+  constexpr size_t k_DefaultBlocksize = 1048576;
+
+  DataArray<T>* dataArray = dynamic_cast<DataArray<T>*>(dataArrayPtr);
+
+  if(dataArray == nullptr)
+  {
+    return k_RbrDANull;
+  }
+
+  const size_t fileSize = std::filesystem::file_size(filename);
+  const size_t numBytesToRead = dataArray->getSize() * sizeof(T);
+  int32_t err = SanityCheckFileSizeVersusAllocatedSize(numBytesToRead, fileSize, skipHeaderBytes);
+
+  if(err < 0)
+  {
+    return k_RbrFileTooSmall;
+  }
+  if(err > 0)
+  {
+    return k_RbrFileTooBig;
+  }
+
+  FILE* f = std::fopen(filename.c_str(), "rb");
+  if(f == nullptr)
+  {
+    return k_RbrFileNotOpen;
+  }
+
+  // Skip some header bytes if the user asked for it.
+  if(skipHeaderBytes > 0)
+  {
+    FSEEK(f, skipHeaderBytes, SEEK_SET);
+  }
+
+  //  std::byte* chunkptr = reinterpret_cast<std::byte*>(dataArray->data());
+  std::byte* chunkptr = reinterpret_cast<std::byte*>(dataArray->template getIDataStoreAs<DataStore<T>>()->data());
+
+  // Now start reading the data in chunks if needed.
+  size_t chunkSize = std::min(numBytesToRead, k_DefaultBlocksize);
+
+  size_t master_counter = 0;
+  while(master_counter < numBytesToRead)
+  {
+    size_t bytes_read = std::fread(chunkptr, sizeof(std::byte), chunkSize, f);
+    chunkptr += bytes_read;
+    master_counter += bytes_read;
+
+    size_t bytesLeft = numBytesToRead - master_counter;
+
+    if(bytesLeft < chunkSize)
+    {
+      chunkSize = bytesLeft;
+    }
+  }
+
+  if(endian == k_EndianCheck)
+  {
+    dataArray->byteSwapElements();
+  }
+
+  std::fclose(f);
+
+  return k_RbrNoError;
+}
+} // namespace
+
+using namespace complex;
+
+RawBinaryReader::RawBinaryReader(DataStructure& dataStructure, RawBinaryReaderInputValues* inputValues, const IFilter* filter, const IFilter::MessageHandler& mesgHandler)
+: m_DataStructure(dataStructure)
+, m_InputValues(inputValues)
+, m_Filter(filter)
+, m_MessageHandler(mesgHandler)
+{
+}
+
+RawBinaryReader::~RawBinaryReader() noexcept = default;
+
+Result<> RawBinaryReader::operator()()
+{
+  return execute();
+}
+
+// -----------------------------------------------------------------------------
+Result<> RawBinaryReader::execute()
+{
+  IDataArray* binaryIDataArray = m_DataStructure.getDataAs<IDataArray>(m_InputValues->createdAttributeArrayPathValue);
+  if(binaryIDataArray == nullptr)
+  {
+    return MakeErrorResult(k_RbrDAError, "Can't obtain DataArray from path '" + m_InputValues->createdAttributeArrayPathValue.toString() + "'.");
+  }
```

Comment:
Since this should be checked in preflight get can use `getDataRefAs` instead which will throw on failure.

```suggestion
  IDataArray& binaryIDataArray = m_DataStructure.getDataRefAs<IDataArray>(m_InputValues->createdAttributeArrayPathValue);
```